getln2.c

```diff
@@ -26,6 +26,8 @@ int sep;
     if (i < n) { substdio_SEEK(ss,*clen = i + 1); *cont = x; return 0; }
     if (!stralloc_readyplus(sa,n)) return -1;
     i = sa->len;
-    sa->len = i + substdio_get(ss,sa->s + i,n);
+    m = substdio_get(ss,sa->s + i,n);
+    if (m != -1)
+      sa->len = i + m;
```

Comment:
Old bug newly spotted during refactoring? Wondering how you came to discover this, as usual, so I can get better at spotting bugs. But also it looks like nothing in-tree calls `getln2()`. Do patches rely on it, or could we delete?

Reply:
if substdio_feed() has been called before substdio_get(), that condition will not happen. In getln2.c, substdio_feed() is getting called before substdio_get(). Even though that condition will not occur, it is a good practice to to check if substdio_get() returned -1.

Reply:
This was spotted during the "read may return -2 and actually mean a very large size" refactoring.

Reply:
Guessing we should hit `getln2()` with `deprecated.h`

Reply:
look closer: getln2() is used in getln().

Reply:
Doh! Maybe we should put it (as `static`) in `getln.c`. But unimportant for this PR.